### PR TITLE
OCPBUGSM-13814: Added SSH public key validation

### PR DESCRIPTION
**Pull request by RazRegev**
In register & update cluster, if user passed SSH key - verify it
is not malformed before updating db.

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -286,6 +286,12 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
 
+	if sshPublicKey := swag.StringValue(&cluster.SSHPublicKey); sshPublicKey != "" {
```
Comment (avishayt) [inline internal/bminventory/inventory.go:289]: cluster.SSHPublicKey is not a *string, no need for StringValue

File: internal/cluster/validations/validations.go [hunk 2]
```diff
@@ -210,3 +214,10 @@ func ValidateNoProxyFormat(noProxy string) error {
 	}
 	return nil
 }
+
+func ValidateSSHPublicKey(sshPublicKey string) (err error) {
+	if _, _, _, _, err = ssh.ParseAuthorizedKey([]byte(sshPublicKey)); err != nil {
+		err = errors.Errorf("Malformed SSH key: %s", sshPublicKey)
```
Comment (avishayt) [inline internal/cluster/validations/validations.go:220]: Nothing useful to print from 'err'?

**Review by avishayt — COMMENTED** [review]
A couple comments inline

**Review by eranco74 — APPROVED** [review]
/lgtm
/approve

File: internal/cluster/validations/validations.go (hunk 2 above)
Comment (RazRegev) [reply internal/cluster/validations/validations.go:220]: no. the only error it can throw is 'ssh: not found'.

**Pull request merged**